connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -97,7 +98,11 @@ public <T> HttpResponse<T> httpRequest(String url, String method, HttpHeaders he
     public <T> HttpResponse<T> httpRequest(String url, String method, HttpHeaders headers, Object requestBodyData,
                                                   TypeReference<T> responseFormat,
                                                   SecretKey sessionKey, String requestSignatureAlgorithm) {
-        HttpClient client = httpClient();
+        // Only try to load SSL configs if we have to (see KAFKA-14816)
+        SslContextFactory sslContextFactory = url.startsWith("https://")
+                ? SSLUtils.createClientSideSslContextFactory(config)
+                : null;
+        HttpClient client = httpClient(sslContextFactory);
```

Comment:
Not related to this change, but any idea why we're creating and starting an HTTP client for every single request here?

Reply:
We wanted to be conservative about unintended changes in behavior, and weren't certain about the thread safety of the `HttpClient` class. Discussion thread [here](https://github.com/apache/kafka/pull/12828#discussion_r1020382238).